Python/tracemalloc.c

```diff
@@ -1258,6 +1260,10 @@ _PyTraceMalloc_TraceRef(PyObject *op, PyRefTracerEvent event,
                         void* Py_UNUSED(ignore))
 {
     if (event != PyRefTracer_CREATE) {
+        /* we don't want bother here with the lock for performance reasons */
+        if (_Py_atomic_load_int32_relaxed(&tracemalloc_config.tracing)) {
+            _Py_atomic_add_ssize((Py_ssize_t *)&tracemalloc_refs_destroyed, 1);
```

Comment:
We should avoid the cast of `size_t *` to `Py_ssize_t *`. Either:

* Use `Py_ssize_t` for the counters. It's okay to use `Py_ssize_t` for values that are never negative.
* Move the operations within the lock so that we don't need atomic operations.
* Add atomic bindings for `size_t`. (I don't think it's worth doing this right now.)

I would lean towards using `Py_ssize_t`.